### PR TITLE
license: re-license to MIT

### LINKED ISSUE #61: Choice of software license
I didn't notice before that the current license currently is EUPL (European Union Public License). This is a weak-copyleft license, and it seems to prevent potentially incorporating code from this package in Meson itself (which is Apache 2.0), or any BSD/MIT licensed package (that includes most of the PyData ecosystem).

Is there a reason for this choice of license @FFY00?

### PR DESCRIPTION
**Pull request by FFY00**
All external contributions have been fairly trivial, but doesn't hurt to still ask, @perigoso and @rgommers do you agree with the relicensing?

Resolves #61

### CODE REVIEW

**Comment by perigoso** [comment]
That is fine with me

@rgommers they did do that

**Review by perigoso — APPROVED** [review]

File: LICENSE [hunk 1]
```diff
@@ -1,287 +1,21 @@
...
-  Reciprocity (LiLiQ-R+).
-
-The European Commission may update this Appendix to later versions of the above
-licences without producing a new version of the EUPL, as long as they provide
-the rights granted in Article 2 of this Licence and protect the covered Source
-Code from exclusive appropriation.
-
-All other changes or additions to this Appendix require the production of a new
-EUPL version.
+Copyright © 2019 Quansight Labs <flains@wuansight.com>
```
Comment (rgommers) [inline LICENSE:1]: typo here in the email. and also, `2019` -> `2021-2022`?

Comment (rgommers) [reply LICENSE:1]: Or just `2021`; easier for maintenance

**Review by rgommers — APPROVED** [review]
> @rgommers they did do that

oops indeed, I misread the diff

**Pull request merged**